packages/web/src/pages/search-page-v2/SearchFilters.tsx

```diff
@@ -216,33 +317,134 @@ const BpmFilter = () => {
           anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
           transformOrigin={{ vertical: 'top', horizontal: 'left' }}
         >
-          <Paper mt='s' border='strong' shadow='far'>
+          <Paper mt='s' border='strong' shadow='far' css={{ width: 242 }}>
             <Flex
-              p='s'
+              w='100%'
+              pv='s'
+              gap='s'
               direction='column'
               alignItems='flex-start'
               role='listbox'
             >
-              <Flex direction='column' w='100%' gap='s'>
+              <Flex
+                direction='column'
+                w='100%'
+                ph='s'
+                onClick={(e) => e.stopPropagation()}
+              >
                 <SegmentedControl
                   options={[
                     { key: 'range', text: 'Range' },
                     { key: 'target', text: 'Target' }
                   ]}
                   selected={bpmFilterType}
-                  onSelectOption={setBpmFilterType}
-                />
-                <TextInput
-                  placeholder={messages.bpm}
-                  label={messages.bpm}
-                  onClick={(e) => {
-                    e.stopPropagation()
-                  }}
-                  onChange={(e) => {
-                    handleBpmInputChange(handleChange)(e.target.value)
+                  onSelectOption={(type) => {
+                    if (type === 'target') {
+                      minRef.current = ''
+                      maxRef.current = ''
+                    } else {
+                      targetBpmRef.current = ''
+                    }
+                    setBpmFilterType(type)
                   }}
                 />
               </Flex>
+              <Divider css={{ width: '100%' }} />
+              {bpmFilterType === 'range' ? (
+                <>
+                  <Flex direction='column' w='100%' ph='s'>
+                    <FilterButtonOptions
+                      options={bpmOptions}
+                      onChange={(option) => {
+                        handleChange(
+                          option.value,
+                          option.helperText ?? option.value
+                        )
+                      }}
+                    />
+                  </Flex>
+                  <Flex
+                    ph='s'
+                    gap='xs'
+                    w={240}
+                    alignItems='center'
+                    onClick={(e) => e.stopPropagation()}
+                  >
+                    <TextInput
+                      label={messages.minBpm}
+                      placeholder={messages.minBpm}
+                      hideLabel
+                      onChange={(e) => {
+                        minRef.current = e.target.value
+                        handleRangeChange(handleChange)
+                      }}
+                      inputRootClassName={css({
```

Comment:
just checking why we have this one-off style?

Reply:
Yea for sure, I have a note and was planning on removing this. this was bc there was no harmony input style that conformed to the design. Sebastian chatted with Sammie about this a bit so I'll check in with him to see what's up there